add new function called detect_color_card

HaleySchuhl opened the pull request:
**Describe your changes**


**Type of update**
Is this a:
* New feature or feature enhancement
* Work in progress

**Associated issues**
#1356 

**Additional context**
Now creates a labeled mask, rather than making users also do the `create_color_card_mask` step. 

**For the reviewer**
See [this page](https://plantcv.readthedocs.io/en/latest/pr_review_process/) for instructions on how to review the pull request.
- [ ] PR functionality reviewed in a Jupyter Notebook
- [ ] All tests pass
- [ ] Test coverage remains 100%
- [ ] Documentation tested
- [ ] New documentation pages added to `plantcv/mkdocs.yml`
- [ ] Changes to function input/output signatures added to `updating.md`
- [ ] Code reviewed
- [ ] PR approved

nfahlgren reviewed (changes requested):
Two general quick comments: 1) deepsource has flagged some changes that need to be addressed; 2) Python general style preference, that we follow in plantcv, is to have all lowercase variables (rather than CamelCase). So like `filtered_contours` instead of `filteredContours`

HaleySchuhl commented:
Thanks for your feedback @nfahlgren . I addressed some of the deepsource stuff already yesterday, just addressed a few more now, but this one in particular I'm not sure how to get around. https://app.deepsource.com/gh/danforthcenter/plantcv/run/01f1e5c0-dd15-45fc-9b6c-1a5e82421e2a/python/PYL-E1133?listindex=0

HaleySchuhl commented:
Probably want to add **kwargs so that hardcoded segmentation & filtering parts of the algorithm could be tweaked.
